### PR TITLE
Use collectd record's host field in perf,unless if value is localhost

### PR DESCRIPTION
**Pull request by sugr4**
@Microsoft/omsagent-devs

### CODE REVIEW

File: source/code/plugins/collectd_lib.rb [hunk 1]
```diff
@@ -27,7 +27,14 @@ def transform(record, hostname)
       data_items = []
       data_info = {}
       data_info["Timestamp"] = record["Timestamp"]
-      data_info["Host"] = hostname
+
+      host = record["host"] if !record["host"].nil?
```
Comment (NarineM) [inline source/code/plugins/collectd_lib.rb]: Why "host" starts with lower case in record["host"]  but upper case in data_info? Are these case sensitive?

Comment (sugr4) [inline source/code/plugins/collectd_lib.rb]: The automatically generated collectd record has keys starting with lowercase. In the perf blob, we just use the convention that the field names start with a capital letter.

**Comment by NarineM** [comment]
I can approve this PR if there is no additional fix or change needed for the parameter 'hostname' when it is localhost or 127.0.0.1.

**Comment by sugr4** [comment]
Deleted the converted properties

**Comment by robbiezhang** [comment]
signoff

**Pull request merged**